plugins/generic/usageEvent/UsageEventPlugin.inc.php

```diff
@@ -286,6 +286,18 @@ function _buildUsageEvent($hookName, $args) {
 			$request, null, $canonicalUrlPage, $canonicalUrlOp, $canonicalUrlParams
 		);
 
+		// Make sure we log the server name and not aliases.
+		$configBaseUrl = Config::getVar('general', 'base_url');
+		$requestBaseUrl = $request->getBaseUrl();
+		if ($requestBaseUrl !== $configBaseUrl) {
+			// Make sure it's not an url override (no alias on that case).
+			if (!in_array($requestBaseUrl, Config::getContextBaseUrls()) && 
+					$requestBaseUrl !== Config::getVar('general', 'base_url[index]')) {
```

Comment:
I'm struggling with this logic as well.  $_SERVER['HTTP_HOST'] will be reflected in $requestBaseUrl, right?  And if the requested HTTP_HOST doesn't match any of the known base_url[] entries, we set the canonical URL to config.inc.php's general base_url?  That doesn't sound right.

If HTTP_HOST doesn't match a known base_url[] entry, we need to look up the base_url[] entry via the current journal's shortname.